### PR TITLE
support an array constructor

### PR DESCRIPTION
**Pull request by kovrus**

### CODE REVIEW

**Comment by kovrus** [comment]
@seut can you please prereview the pr?

**Comment by kovrus** [comment]
@matriv can you please preview? documentation is already acked by the core team

File: sql/src/test/java/io/crate/operation/scalar/arithmetic/ArrayFunctionTest.java [hunk 1]
```diff
@@ -22,19 +22,49 @@
...
+import io.crate.analyze.symbol.Literal;
+import io.crate.operation.scalar.AbstractScalarFunctionsTest;
 import io.crate.types.DataType;
 import io.crate.types.DataTypes;
 import org.junit.Test;
 
 import java.util.Arrays;
 
-public class ArrayFunctionTest extends CrateUnitTest {
+import static io.crate.testing.TestingHelpers.isLiteral;
```
Comment (matriv) [inline sql/src/test/java/io/crate/operation/scalar/arithmetic/ArrayFunctionTest.java]: please check and re-arange imports

**Comment by kovrus** [comment]
@dobe can you please review?

File: CHANGES.txt [hunk 2]
```diff
@@ -2,6 +2,9 @@
 Changes for Crate
 =================
 
+ - Added support of expressions in the array constructors. Allowed to
```
Comment (chaudum) [inline CHANGES.txt]: the entry belongs underneath `Unreleased` section

**Pull request merged**